File: src/ELibrary.Frontend/src/app/modules/shared/domain/dtos/library/book/createBookRequest.ts
import { Book, CoverType } from "../../../..";

export interface CreateBookRequest {
    name: string;
    publicationDate: Date;
    price: number;
    coverType: CoverType;
    pageAmount: number;
    coverImgUrl: string,
    authorId: number;
    genreId: number;
    publisherId: number;
}
export function mapBookToCreateBookRequest(book: Book): CreateBookRequest {
    return {
        name: book.name,
        publicationDate: book.publicationDate,
        price: book.price,
        coverType: book.coverType,
        pageAmount: book.pageAmount,
        coverImgUrl: book.coverImgUrl,
        authorId: book.author.id,
        genreId: book.genre.id,
        publisherId: book.publisher.id
    }
}

File: src/ELibrary.Backend/LibraryApi/Domain/Dtos/Book/CreateBookRequest.cs
using LibraryShopEntities.Domain.Entities.Library;

namespace LibraryApi.Domain.Dto.Book
{
    public class CreateBookRequest
    {
        public string? Name { get; set; }
        public DateTime PublicationDate { get; set; }
        public decimal Price { get; set; }
        public CoverType CoverType { get; set; }
        public string? CoverImgUrl { get; set; }
        public string? Description { get; set; }
        public int PageAmount { get; set; }
        public int AuthorId { get; set; }
        public int GenreId { get; set; }
        public int PublisherId { get; set; }
    }
}